### PR TITLE
[plugin-web-app] Allow to use locators in video steps

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-plugin-web-app/src/main/resources/steps/defaults/video.steps [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+!-- Step is deprecated and will be removed in VIVIDUS 0.5.0. Use instead: When I rewind time to `$number` seconds in video player located `$locator`
+When I rewind time to `<number>` seconds in video player located `xpath(//video[@*='<videoPlayerName>'])`
+
+Composite: When I play video in the video player with the name '$videoPlayerName'
+!-- Step is deprecated and will be removed in VIVIDUS 0.5.0. Use instead: When I play video in video player located `$locator`
+When I play video in video player located `xpath(//video[@*='<videoPlayerName>'])`
+
+Composite: When I pause video in the video player with the name '$videoPlayerName'
+!-- Step is deprecated and will be removed in VIVIDUS 0.5.0. Use instead: When I pause video in video player located `$locator`
+When I pause video in video player located `xpath(//video[@*='<videoPlayerName>'])`
```
Comment (ikalinin1) [inline vividus-plugin-web-app/src/main/resources/steps/defaults/video.steps:11]: Remove deprecated video steps

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 2]
```diff
@@ -689,3 +689,69 @@ where the first measure is window width, the last one is window height.
...
+* `$variableName` - The variable name to save the info. If the variable name is *info*, the following variables will be created:
+** `${info.src}`- The the current source of the audio/video element
+** `${info.duration}` - The length of the current audio/video (in seconds)
+** `${info.currentTime}` - The current playback position in the audio/video(in seconds)
+** `${info.networkState}` - The the current network state of the audio/video. For more information see: https://www.w3schools.com/tags/av_prop_networkstate.asp[Network State]
+
+.Video player verifications
+[source,gherkin]
+----
+Given I am on a page with the URL 'https://example.com/'
```
[the remark this answers is not in the document]
Comment (ikalinin1) [reply docs/modules/plugins/pages/plugin-web-app.adoc:748]: image

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 3]
```diff
@@ -689,3 +689,69 @@ where the first measure is window width, the last one is window height.
...
+When I pause video in video player located `cssSelector(video)`
+When I rewind time to `777` seconds in video player located `cssSelector(video)`
+When I play video in video player located `cssSelector(video)`
+When I save info from video player located `cssSelector(video)` to SCENARIO variable `info`
+Then `${info.duration}` is > `1000`
+Then `${info.duration}` is > `1000`
```
Comment (uarlouski) [inline docs/modules/plugins/pages/plugin-web-app.adoc:754]: info.duration gets checked two times

File: vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/VideoPlayerActions.java [hunk 4]
```diff
@@ -41,4 +41,15 @@ public void pause(WebElement videoPlayer)
     {
         javascriptActions.executeScript("arguments[0].pause();", videoPlayer);
     }
+
+    @Override
+    public Object getInfo(WebElement videoPlayer)
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/VideoPlayerActions.java:46]:
```suggestion
    public Map<String, Object> getInfo(WebElement videoPlayer)
```
?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**